Sources/Tracking/Util/Log.swift

```diff
@@ -1,34 +1,96 @@
 import Foundation
+#if canImport(os)
+import os.log
+#endif
 
+/// mockable logger + abstract that allows you to log to multiple places if you wish
 public protocol Logger: AutoMockable {
-    func verbose(_ message: String)
+    /// the noisey log level. Feel free to spam this log level with any
+    /// information about the SDK that would be useful for debugging the SDK.
     func debug(_ message: String)
+    /// Not noisy log messages. Good for general information such as
+    /// when the background queue begins and ends running but use `debug`
+    /// for the status of each background queue task running.
     func info(_ message: String)
-    func warning(_ message: String)
+    /// the SDK is in an unstable state that you want to notify
+    /// the customer or our development team about.
     func error(_ message: String)
 }
 
+/// none - no logs will be made
+/// error - only log when there is an error in the SDK (default)
+/// info - basic SDK informion. Somewhat noisy. Recommended to start debugging SDK.
+/// debug - most noisy. See all of the logs made from the SDK.
+public enum CioLogLevel {
+    case none
+    case error
+    case info
+    case debug
+
+    #if canImport(os)
+    func shouldLog(_ level: OSLogType) -> Bool {
+        switch self {
+        case .none: return false
+        case .error:
+            return level == .error
+        case .info:
+            return level == .error || level == .info
+        case .debug:
+            return true
+        }
+    }
+    #endif
+}
+
+// log messages to console.
 // sourcery: InjectRegister = "Logger"
 public class ConsoleLogger: Logger {
-    private let prefix = "[Cio]"
+    // allows filtering in Console mac app
+    private let logSubsystem = "io.customer.sdk"
+    private let logCategory = "CIO"
+
+    private let sdkConfigStore: SdkConfigStore
 
-    public func verbose(_ message: String) {
-        print("\(prefix) Verbose: \(message)")
+    private var minLogLevel: CioLogLevel {
+        sdkConfigStore.config.logLevel
     }
 
-    public func debug(_ message: String) {
-        print("\(prefix) Debug: \(message)")
+    init(sdkConfigStore: SdkConfigStore) {
+        self.sdkConfigStore = sdkConfigStore
     }
 
-    public func info(_ message: String) {
-        print("\(prefix) Info: \(message)")
+    #if canImport(os)
+    // Unified logging for Swift. https://www.avanderlee.com/workflow/oslog-unified-logging/
+    // This means we can view logs in xcode console + Console app.
+    private func printMessage(_ message: String, _ level: OSLogType) {
+        if !minLogLevel.shouldLog(level) { return }
+
+        if #available(iOS 14, *) {
+            let logger = os.Logger(subsystem: self.logSubsystem, category: self.logCategory)
+            logger.info("\(message, privacy: .public)")
```

Comment:
how come we're not using the os.Logger / OSLog levels here, and doing everything at .info? `level` seems unused?